bson/unmarshal.go

```diff
@@ -61,6 +61,21 @@ func UnmarshalWithContext(dc bsoncodec.DecodeContext, data []byte, val interface
 	return unmarshalFromReader(dc, vr, val)
 }
 
+// UnmarshalValue parses the BSON value of type t with bson.DefaultRegistry and
+// stores the result in the value pointed to by val. If val is nil or not a pointer,
+// UnmarshalValue returns an error.
+func UnmarshalValue(t bsontype.Type, data []byte, val interface{}) error {
```

Comment:
The addition of "UnmarshalValue" is scheduled as a 2.0 improvement in [GODRIVER-1892](https://jira.mongodb.org/browse/GODRIVER-1892). I don't think we should add it here unless GODRIVER-2779 is marked a 2.0 and 1892 a duplicate.

For the sake of this ticket, perhaps it would be best to make these unexported test helpers. We could mark them as "TODO: GODRIVER-1892 remove with the introduction of stable API UnmarshalValue".

@matthewdale Do you have thoughts on this?

Reply:
GODRIVER-1892 is part of the Go Driver 2.0 preparation epic, which is a collection of non-breaking changes we want to make in Go Driver 1.x that makes transition to Go Driver 2.0 easier. We can make this additive API change in any Go Driver 1.x release, which will remain part of the Go Driver 2.0 API.

TLDR: We should keep this change because it is strictly additive and should be part of the Go Driver 1.x and 2.x API.